### PR TITLE
Stripe: Migrate from /refund to /refunds

### PR DESCRIPTION
**Pull request by matthelm**
This PR updates the Stripe endpoint we use for creating refunds. The previous endpoint returned the Charge object, while the new endpoint returns the Refund object, so as a result the test and fixtures have been updated. Will this change be a problem for clients using Active Merchant?

This PR also includes the metadata object in the `/refunds` request, which is added in this new endpoint.

*Before*
`https://api.stripe.com/v1/charges/{CHARGE_ID}/refund`

*After*
`https://api.stripe.com/v1/charges/{CHARGE_ID}/refunds`

### CODE REVIEW

**Comment by matthelm** [comment]
@AnotherJoSmith mind giving this an initial review?

File: lib/active_merchant/billing/gateways/stripe.rb [hunk 1]
```diff
@@ -115,9 +115,12 @@ def refund(money, identification, options = {})
         add_amount(post, money, options)
         post[:refund_application_fee] = true if options[:refund_application_fee]
         post[:reverse_transfer] = options[:reverse_transfer] if options[:reverse_transfer]
+        post[:metadata] = options[:metadata]
+
+        options.delete(:metadata)
```
Comment (AnotherJoSmith) [inline lib/active_merchant/billing/gateways/stripe.rb]: Why do you need to delete the metadata from the options map here?

**Comment by AnotherJoSmith** [comment]
Looks good :monorail:

**Comment by matthelm** [comment]
Please review @j-mutter @aprofeit

File: test/unit/gateways/stripe_test.rb [hunk 2]
```diff
@@ -342,8 +342,7 @@ def test_successful_refund
     assert response = @gateway.refund(@refund_amount, 'ch_test_charge')
     assert_success response
 
-    assert_equal 'ch_test_charge', response.authorization
-    assert response.test?
```
Comment (j-mutter) [inline test/unit/gateways/stripe_test.rb]: Eh, not a `test` anymore? Or you don't feel this is necessary?

**Comment by j-mutter** [comment]
Remote tests?

File: test/unit/gateways/stripe_test.rb (hunk 2 above)
Comment (matthelm) [inline test/unit/gateways/stripe_test.rb]: Stripe's Refund object doesn't have a `livemode` flag to use when populating `test?`. Suggestions?
https://stripe.com/docs/api#refund_object

Comment (j-mutter) [inline test/unit/gateways/stripe_test.rb]: Oh ok - TIL. I guess it just uses the associated charge_id to figure it out. This is fine, then

**Comment by matthelm** [comment]
@j-mutter I ran the remote tests each on their own. The output is [here](https://gist.github.com/matthelm/0bbc060740577d009c0c). Many tests failed in the first three scenarios, but they were also failing on master. All the tests in `remote_stripe_test.rb` were green. Is there anything else I need to test?

```
remote_stripe_apple_pay_test.rb
remote_stripe_connect_test.rb
remote_stripe_emv_test.rb
remote_stripe_test.rb
```

**Comment by j-mutter** [comment]
:+1:

**Comment by ntalbott** [comment]
@matthelm @j-mutter I was surprised this didn't require a change/bump to the default Stripe API version? https://github.com/activemerchant/active_merchant/blob/e28815158657c312bd6de1d08867f4bd22dc0f59/lib/active_merchant/billing/gateways/stripe.rb#L426

**Comment by j-mutter** [comment]
I'm pretty sure this endpoint predates the current minimum. @matthelm can you confirm?

**Comment by matthelm** [comment]
@ntalbott I ran the remote tests, ensuring the default value `2015-04-07` was used, and all the tests were green. The last file in the gist is the test output. https://gist.github.com/matthelm/0bbc060740577d009c0c#file-remote_stripe_test-rb

Is there anything else you'd like me to do to verify this is sound?

**Comment by ntalbott** [comment]
Nope, sounds good. So `2015-04-07` must allow either/or, eh?

**Comment by matthelm** [comment]
Exactly

File: test/unit/gateways/stripe_test.rb [hunk 3]
```diff
@@ -1396,44 +1396,16 @@ def successful_partially_refunded_response(options = {})
   def successful_void_response
     <<-RESPONSE
     {
-      "id": "ch_4IrhQMqukqu7C2",
-      "object": "charge",
-      "created": 1403816613,
-      "livemode": false,
-      "paid": true,
-      "amount": 50,
+      "id": "re_173VMpAWOtgoysogOSE7Hzss",
```
Comment (girasquid) [inline test/unit/gateways/stripe_test.rb]: voids create refund objects?

**Comment by girasquid** [comment]
One question, but looks ok.

File: test/unit/gateways/stripe_test.rb (hunk 3 above)
Comment (matthelm) [inline test/unit/gateways/stripe_test.rb]: Stripe's AM adapter implements the `void` method using `/refund`, so wouldn't it be creating refunds?

Comment (girasquid) [inline test/unit/gateways/stripe_test.rb]: Yup, you're right. Carry on.

**Comment by matthelm** [comment]
I'm ready to merge, but am not an "Owner". Is somebody able to merge for me?

**Comment by girasquid** [comment]
Done.

**Pull request closed without merging**